target/scripts/check-for-changes.sh

```diff
@@ -7,6 +7,8 @@ LOG_DATE=$(date +"%Y-%m-%d %H:%M:%S ")
 _notify 'task' "${LOG_DATE} Start check-for-changes script."
 
 SCRIPT_NAME="$(basename "$0")"
+LOCK_ID="${RANDOM}" # Required for create_lock uniqueness and cleanup
+trap "remove_lock ${SCRIPT_NAME} ${LOCK_ID}" EXIT
```

Comment:
This is only case where a `create_lock` line is avoided from the prior pattern used. I understand that is because unlike other cases, `create_lock` is being called in a loop below instead and only if a change is detected.

The comment for `LOCK_ID` is not on the other definitions for it: `# Required for create_lock uniqueness and cleanup`. To clarify this is for uniqueness of the `create_lock` below from other locks right? The `create_lock` below does not need to be unique if called more than once in that loop, it is intended to reuse the same `LOCK_ID` defined here?

Your commit messages and PR description do not clarify why the `create_lock` and `remove_lock` were moved into the conditional `[$? -eq 1 ]`?

Reply:
> Your commit messages and PR description do not clarify why the create_lock and remove_lock were moved into the conditional [$? -eq 1 ]?

Added, thanks!

> The comment for LOCK_ID is not on the other definitions for it: # Required for create_lock uniqueness and cleanup. To clarify this is for uniqueness of the create_lock below from other locks right? The create_lock below does not need to be unique if called more than once in that loop, it is intended to reuse the same LOCK_ID defined here?

I don't know what I was thinking actually... Moved the lock ID into the helper